tests/codegen/reg-struct-return.rs

```diff
@@ -0,0 +1,64 @@
+// Checks how `reg-struct-return` flag works with different calling conventions:
+// Return struct with 8/16/32/64 bit size will be converted into i8/i16/i32/i64
+// (like abi_return_struct_as_int target spec).
+// x86 only.
+
+//@ compile-flags: --target i686-unknown-linux-gnu -Zreg-struct-return -O -C no-prepopulate-passes
+//@ needs-llvm-components: x86
+
+#![crate_type = "lib"]
+#![no_std]
+#![no_core]
+#![feature(no_core, lang_items)]
+
+#[lang = "sized"]
+trait Sized {}
+#[lang = "copy"]
+trait Copy {}
+
+#[repr(C)]
+pub struct Foo {
+    x: u32,
+    y: u32,
+}
+
+pub mod tests {
+    use Foo;
+
+    // CHECK: i64 @f1()
+    #[no_mangle]
+    pub extern "fastcall" fn f1() -> Foo {
+        Foo { x: 1, y: 2 }
+    }
+
+    // reg-struct-return doesn't work for "Rust" calling conv
+    // CHECK: { i32, i32 } @f2()
+    #[no_mangle]
+    pub extern "Rust" fn f2() -> Foo {
+        Foo { x: 1, y: 2 }
+    }
```

Comment:
Why doesn't it work for the Rust calling convention? Is that what we want? I assume that there's a reason we use this flag, but I don't know whether it's just a question of performance or if it's an important mitigation.

@andyhhp Do you know if this matters?

Reply:
At technical level, "external" abi's are processed a little in different way (another place in code). It is not a big problem to support the same behavior for "Rust" calling conv, I just need to know if it is required.

Reply:
> Why doesn't it work for the Rust calling convention? Is that what we want? I assume that there's a reason we use this flag, but I don't know whether it's just a question of performance or if it's an important mitigation.
> 
> @andyhhp Do you know if this matters?

Rust and C need to agree on how to return >GPR sized values, if this happens to be used in either direction.

Whether that is via some bindings using the Rust calling conventions is something I can't answer.

The use of >GPR sized return values is rare in C because it's irritating to set up.  You need a struct/typedef with a fairly precise layout, but the code generation improvements are substantial, hence why they do get used occasionally.

Reply:
The Rust ABI is not stable. C should not call an `extern "Rust" fn`. This _matters_ for `extern "C" fn` and a few others. And reusing this code, tuned for `extern "C"` and other foreign ABIs, for Rust ABI code? Proved incapable of compiling libcore in #130432.

We're not going to be scope-creeping this PR into `extern "Rust"`. _That_ can be addressed alongside the *extensive* cleanups of our ABI code that it will require.